docs/build-and-push-assets.md

```diff
@@ -114,6 +115,32 @@ Encore.cleanupOutputBeforeBuild(['*.js', '*.css'])
 
 ---
 
+> Can I decide when to run `COMPILE_SCRIPT_PROD` or `COMPILE_SCRIPT_DEV`?
+ 
+Yes, you can define the `inputs.MODE` which is either `dev` or `prod`. Depending on the set value the corresponding script will be executed.
+
+When left empty, the default logic is applied. So we can define following scenarios:
+
+| MODE     | scenario            | script                |
+|----------|---------------------|-----------------------|
+| `''`     | push to branch.     | `COMPILE_SCRIPT_DEV`  |
+| `''`     | create release/tag. | `COMPILE_SCRIPT_PROD` |
+| `'dev'`  | _not evaluated_     | `COMPILE_SCRIPT_DEV`  |
+| `'prod'` | _not evaluated_     | `COMPILE_SCRIPT_PROD` |
+
+**Example:** I want to push to a branch `production` and "production"-ready assets have to be compiled:
```

Comment:
Isn't the scenario described in the example the default behavior, so `inputs.MODE` wouldn't be needed?
```suggestion
> Can I decide when to run `COMPILE_SCRIPT_PROD` or `COMPILE_SCRIPT_DEV`?
 
Use the `inputs.MODE` and set it to `dev` or `prod`. Depending on the value, the corresponding script will be executed. When left empty, the default logic is applied.

The following table provides an overview when `COMPILE_SCRIPT_DEV` or `COMPILE_SCRIPT_PROD` is used:

| MODE   | scenario           | script                |
|--------|--------------------|-----------------------|
| `''`   | push to branch     | `COMPILE_SCRIPT_DEV`  |
| `''`   | create release/tag | `COMPILE_SCRIPT_PROD` |
| `dev`  | _not evaluated_    | `COMPILE_SCRIPT_DEV`  |
| `prod` | _not evaluated_    | `COMPILE_SCRIPT_PROD` |

**Example:** I want to push to a branch `production` and "production"-ready assets should be compiled:
```